BankManagerUIApp/BankManagerUIApp/Model/CustomerQueue.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  CustomerQueue.swift
+//  BankManagerUIApp
+//
+//  Created by Mangdi, Mene on 31/10/2022.
+//
+
+import Foundation
+
+struct CustomerQueue<T> {
+    var front: Node<T>?
+    var rear: Node<T>?
+    
+    var isEmpty: Bool {
+        rear == nil ? true : false
+    }
+    
+    mutating func enqueue(data: T?) {
+        let newNode = Node(data: data, next: nil)
+        
+        if isEmpty {
+            front = newNode
+            rear = newNode
+        } else {
+            rear?.next = newNode
+            rear = newNode
+        }
+    }
+    
+    mutating func dequeue() -> T? {
+        if isEmpty {
+            return nil
+        } else {
+            let data = front?.data
+            front = front?.next
+            return data
+        }
+    }
```

Comment:
옵셔널을 적극 활용해주세요.
굳이 isEmpty를 체크해야할까요?

```suggestion
    mutating func dequeue() -> T? {
        let data = front?.data
        front = front?.next
        
        return data
    }
```

Reply:
리팩토링 하면서 망디와 수정해 보겠습니다. ^^

Reply:
옵셔널을 반환하는메서드는 이런식으로 축약해서 작성할수있군요..!
저번 계산기프로젝트때도 비슷한실수를 했었는데 이번에도 같은실수를 범했네요 ㅠ
다시한번 뇌속에 각인시키는 계기가 되었습니다